OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/string_analysis/interpretation/AbstractStringInterpreter.scala

```diff
@@ -0,0 +1,213 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj.tac.fpcf.analyses.string_analysis.interpretation
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+import org.opalj.fpcf.Entity
+import org.opalj.fpcf.EOptionP
+import org.opalj.fpcf.Property
+import org.opalj.fpcf.PropertyStore
+import org.opalj.value.ValueInformation
+import org.opalj.br.cfg.CFG
+import org.opalj.br.Method
+import org.opalj.br.analyses.DeclaredMethods
+import org.opalj.br.DefinedMethod
+import org.opalj.br.fpcf.properties.StringConstancyProperty
+import org.opalj.br.fpcf.properties.cg.Callees
+import org.opalj.br.fpcf.properties.string_definition.StringConstancyInformation
+import org.opalj.tac.Stmt
+import org.opalj.tac.TACStmts
+import org.opalj.tac.fpcf.analyses.string_analysis.V
+import org.opalj.tac.TACMethodParameter
+import org.opalj.tac.TACode
+import org.opalj.tac.fpcf.properties.TACAI
+import org.opalj.tac.Assignment
+import org.opalj.tac.DUVar
+import org.opalj.tac.Expr
+import org.opalj.tac.ExprStmt
+import org.opalj.tac.FunctionCall
+import org.opalj.tac.fpcf.analyses.string_analysis.interpretation.interprocedural.InterproceduralInterpretationHandler
+import org.opalj.tac.fpcf.analyses.string_analysis.InterproceduralComputationState
+import org.opalj.tac.fpcf.analyses.string_analysis.NonFinalFunctionArgs
+import org.opalj.tac.fpcf.analyses.string_analysis.NonFinalFunctionArgsPos
+import org.opalj.tac.fpcf.analyses.string_analysis.P
+
+/**
+ * @param cfg The control flow graph that underlies the instruction to interpret.
+ * @param exprHandler In order to interpret an instruction, it might be necessary to interpret
+ *                    another instruction in the first place. `exprHandler` makes this possible.
+ *
+ * @note The abstract type [[InterpretationHandler]] allows the handling of different styles (e.g.,
+ *       intraprocedural and interprocedural). Thus, implementation of this class are required to
+ *       clearly indicate what kind of [[InterpretationHandler]] they expect in order to ensure the
+ *       desired behavior and not confuse developers.
+ *
+ * @author Patrick Mell
+ */
+abstract class AbstractStringInterpreter(
+        protected val cfg:         CFG[Stmt[V], TACStmts[V]],
+        protected val exprHandler: InterpretationHandler
+) {
+
+    type T <: Any
```

Comment:
Any or AnyRef?

Reply:
Actually, `ASTNode[V]` since we handle either expressions or statements. Done in recent commit.